lib/ctcs2_to_junit.pm

```diff
@@ -102,12 +102,12 @@ sub generateXML {
         if (!get_var('QA_TESTSUITE') && ($case_status eq 'failure' || $case_status eq 'skipped')) {
             (my $test_path = $test) =~ s/-/\//;
             $test_path = '/opt/log/' . $test_path;
-            my $test_out_content = script_output("cat $test_path| sed \"s/'//g\"", 600);
+            my $test_out_content = script_output("cat $test_path| sed \"s/'//g\" | sed \"s/[\\x00-\\x08\\x0B-\\x0C\\x0E-\\x1F]//g\"", 600);
             $writer->startTag('system-out');
             $writer->characters($test_out_content);
             $writer->endTag('system-out');
             if ($case_status eq 'failure') {
-                my $test_err_content = script_output("echo '====out.bad log====';if [ -f $test_path.out.bad ];then cat $test_path.out.bad|sed \"s/'//g\";else echo '$test_path.out.bad not exist';fi;echo '====full log====';if [ -f $test_path.full ]; then cat $test_path.full| sed \"s/'//g\"; else echo '$test_path.full not exist';fi;", 600);
+                my $test_err_content = script_output("echo '====out.bad log====';if [ -f $test_path.out.bad ];then cat $test_path.out.bad|sed \"s/'//g\" | sed \"s/[\\x00-\\x08\\x0B-\\x0C\\x0E-\\x1F]//g\";else echo '$test_path.out.bad not exist';fi;echo '====full log====';if [ -f $test_path.full ]; then cat $test_path.full| sed \"s/'//g\" | sed \"s/[\\x00-\\x08\\x0B-\\x0C\\x0E-\\x1F]//g\"; else echo '$test_path.full not exist';fi;", 600);
```

Comment:
What's about at least extracting common part of the string? In general it's hardly readable, I would at least put it as multiline string variable.

Reply:
OK, I'd like to change to multiline, but I wonder if tidy like it or not. Let me have a try.